drivers/crypto/crypto_stm32_sha.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2025 Bayrem Gharsellaoui
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/crypto/crypto.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/reset.h>
+#include <soc.h>
+
+#include "crypto_stm32_sha_priv.h"
+
+#define LOG_LEVEL CONFIG_CRYPTO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(crypto_stm32_sha);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_hash)
+#define DT_DRV_COMPAT st_stm32_hash
+#else
+#error No STM32 HW Hash Accelerator in device tree
+#endif
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_hash)
+#define STM32_HASH_TYPEDEF HASH_TypeDef
+#endif
+
+#define CRYPTO_HASH_MAX_SESSIONS CONFIG_CRYPTO_HASH_MAX_SESSIONS
+
+static struct crypto_stm32_sha_session crypto_stm32_sha_sessions[CRYPTO_HASH_MAX_SESSIONS];
+
+static int crypto_stm32_sha_get_unused_session_index(const struct device *dev)
+{
+	int i;
+	struct crypto_stm32_sha_data *data = CRYPTO_STM32_SHA_DATA(dev);
+
+	k_sem_take(&data->session_sem, K_FOREVER);
+
+	for (i = 0; i < CRYPTO_HASH_MAX_SESSIONS; i++) {
```

Comment:
```suggestion
	for (int i = 0; i < ARRAY_SIZE(crypto_stm32_sha_sessions); i++) {
```

(Moving the `int i` declaration in the `for()` is a nit)

Reply:
Done